### ISSUE TITLE
youtube-dl and mkv4 encoding error

### ISSUE THREAD

**Issue by republicus** [YouPHPTube/YouPHPTube-Encoder#146]
I am have setup multiple encoder systems which appears to fail on ffmpeg attempts at encoding. I have tried both youtube (Import Video) and encoding local mkv videos. I'm not sure what's wrong.

I am attaching the log from the local mkv failed encoding but youtube-dl appears to be very similar:

The encoder shows blank error info: error []
![empty-error](https://user-images.githubusercontent.com/47370709/59153379-f926bf80-8a25-11e9-8367-537260420d4a.PNG)

**Attached log:**
[encode3.log](https://github.com/YouPHPTube/YouPHPTube-Encoder/files/3268994/encode3.log)

**Raw from log:**
root@phptube-encode3:/var/www/html/YouPHPTube-Encoder/videos# tail -f youphptube.log

[08-Jun-2019 22:57:22 UTC] Upload.php will set format`

[08-Jun-2019 22:57:22 UTC] Upload.php will let function decide decideFormatOrder

[08-Jun-2019 22:57:22 UTC] decideFormatOrder: {"audioOnly":"false","spectrum":"false","webm":"false","inputHLS":"false","inputLow":"true","inputSD":"true","inputHD":"true","title":"Sony 4K Demo: Another World","description":"Video 4K Demo by Sony.","categories_id":"1"}

[08-Jun-2019 22:57:22 UTC] decideFormatOrder: MP4 All

[08-Jun-2019 22:57:22 UTC] {"status":"error", "msg":"getDurationFromFile ERROR, File () Not Found"}

[08-Jun-2019 22:57:22 UTC] YouPHPTube-Encoder sending file to https://tube.mydomain.com/youPHPTubeEncoder.json

[08-Jun-2019 22:57:22 UTC] YouPHPTube-Encoder reading file from

[08-Jun-2019 22:57:23 UTC] YouPHPTube-Streamer answer {"error":false,"video_id":34}

[08-Jun-2019 22:57:23 UTC] {"error":false,"format":"mp4","file":"","resolution":"","videoDownloadedLink":null,"target":"https:\/\/tube.mydomain.com\/youPHPTubeEncoder.json","postFields":10,"response_raw":"{\"error\":false,\"video_id\":34}","response":{"error":false,"video_id":34}}

[08-Jun-2019 22:57:23 UTC] downloadFile: start queue_id = 2

[08-Jun-2019 22:57:23 UTC] downloadFile: url = https://encode3.mydomain.com/videos/original_Sony4KDemoAnotherWorld_YPTuniqid_5cfc3d51daf624.41795289

[08-Jun-2019 22:57:23 UTC] downloadFile:strpos global['webSiteRootURL'] = https://encode3.mydomain.com/

[08-Jun-2019 22:57:23 UTC] downloadFile: this file was uploaded from file and thus is in the videos

[08-Jun-2019 22:57:23 UTC] downloadFile: downloadedFile = /var/www/html/YouPHPTube-Encoder/videos/original_Sony4KDemoAnotherWorld_YPTuniqid_5cfc3d51daf624.41795289 | url = https://encode3.mydomain.com/videos/original_Sony4KDemoAnotherWorld_YPTuniqid_5cfc3d51daf624.41795289

[08-Jun-2019 22:57:24 UTC] Try to get UTF8 URL https://encode3.mydomain.com/videos/original_Sony4KDemoAnotherWorld_YPTuniqid_5cfc3d51daf624.41795289

[08-Jun-2019 22:57:24 UTC] Try to get UTF8 decode URL https://encode3.mydomain.com/videos/original_Sony4KDemoAnotherWorld_YPTuniqid_5cfc3d51daf624.41795289

[08-Jun-2019 22:57:24 UTC] downloadFile: success

[08-Jun-2019 22:57:24 UTC] downloadFile: {"error":false,"filename":"2_tmpFile.mp4","pathFileName":"\/var\/www\/html\/YouPHPTube-Encoder\/videos\/2_tmpFile.mp4"}

[08-Jun-2019 22:57:24 UTC] sendImages: Sending image to [34]

[08-Jun-2019 22:57:25 UTC] Duration found: 0:03:11

[08-Jun-2019 22:57:25 UTC] sendImages: YouPHPTube-Encoder sending file to https://tube.mydomain.com/objects/youPHPTubeEncoderReceiveImage.json.php

[08-Jun-2019 22:57:25 UTC] sendImages: YouPHPTube-Encoder reading file from /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4

[08-Jun-2019 22:57:25 UTC] getImage: ffmpeg -ss 00:01:35 -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vframes 1 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4.jpg

[08-Jun-2019 22:57:28 UTC] getImage: takes 2.9223198890686 sec to complete

[08-Jun-2019 22:57:28 UTC] getGifImage

[08-Jun-2019 22:57:28 UTC] getGif: Starts

[08-Jun-2019 22:57:34 UTC] getGif: takes 5.9460439682007 sec to complete

[08-Jun-2019 22:57:39 UTC] sendImages: curl_init

[08-Jun-2019 22:57:39 UTC] sendImages: curl_exec

[08-Jun-2019 22:57:39 UTC] sendImages: YouPHPTube-Streamer answer {"error":false,"video_id":34}

[08-Jun-2019 22:57:39 UTC] {"error":false,"file":"\/var\/www\/html\/YouPHPTube-Encoder\/videos\/2_tmpFile.mp4","target":"https:\/\/tube.mydomain.com\/objects\/youPHPTubeEncoderReceiveImage.json.php","postFields":6,"response_raw":"{\"error\":false,\"video_id\":34}","response":{"error":false,"video_id":34}}

[08-Jun-2019 22:57:39 UTC] run:runMultiResolution

[08-Jun-2019 22:57:39 UTC] YouPHPTube-Encoder Start Encoder [ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:720 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_HD.mp4]

[08-Jun-2019 22:57:39 UTC] ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:720 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_HD.mp4 --- [] --- (8, /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4, /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_HD.mp4, 2)

[08-Jun-2019 22:57:39 UTC] YouPHPTube-Encoder Start Encoder [ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:540 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_SD.mp4]

[08-Jun-2019 22:57:39 UTC] ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:540 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_SD.mp4 --- [] --- (7, /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4, /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_SD.mp4, 2)

[08-Jun-2019 22:57:39 UTC] YouPHPTube-Encoder Start Encoder [ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:360 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_Low.mp4]

[08-Jun-2019 22:57:39 UTC] ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:360 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_Low.mp4 --- [] --- (1, /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4, /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_Low.mp4, 2)

[08-Jun-2019 22:57:39 UTC] Trying again: [1] => Execute code error "Array\n(\n)\n"
 Code: ffmpeg -i /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile.mp4 -vf scale=-2:360 -movflags faststart -preset ultrafast -vcodec h264 -acodec aac -strict -2 -max_muxing_queue_size 1024 -y /var/www/html/YouPHPTube-Encoder/videos/2_tmpFile_converted_Low.mp4

**Comment by republicus** [comment]
This issue is solved. While it is not a bug, the Ubuntu 16.04 install instructions, if not updated, will likely result in these errors.

I located the problem by issuing the ffmpeg command manually.
Doing so resulted in the error: 
`Unrecognized option 'max_muxing_queue_size'`

Which revealed that the version of  ffmpeg on Ubuntu 16.04 is outdated. My installed version was 2+ and I've updated using jonathonf's ffmpeg-4 ppa.

So if you are following the Install instructions for Ubuntu 16.04, more steps are necessary to locally encode.

My install is a minimal setup so some packages may be included but the following instructions resolved my errors:

`apt install software-properties-common`

`add-apt-repository ppa:jonathonf/ffmpeg-4`

`apt-get update && apt-get -y install ffmpeg php-xml python curl && apt-get -y upgrade`
